### PR TITLE
Fix shifting array

### PR DESCRIPTION
**Pull request by drjova**

### CODE REVIEW

File: dictdiffer/utils.py [hunk 1]
```diff
@@ -271,3 +275,24 @@ def are_different(first, second, tolerance):
...
+    """Strip all the values that are instance of ``RemovedObject``."""
+    if isinstance(obj, LIST_TYPES):
+        return [
+            strip_removed_objects(value)
+            for value in obj if not isinstance(value, RemovedObject)
+        ]
+    if isinstance(obj, DICT_TYPES):
+        new_dict = {}
+        for key, value in obj.items():
+            clean_value = strip_removed_objects(value)
```
Comment (david-caro) [inline dictdiffer/utils.py]: You might want to put this inside the if.

File: dictdiffer/utils.py [hunk 2]
```diff
@@ -271,3 +275,24 @@ def are_different(first, second, tolerance):
...
+            if not isinstance(clean_value, RemovedObject):
+                new_dict[key] = clean_value
+        return new_dict
+    return obj
+
+
+class RemovedObject(object):
```
Comment (david-caro) [inline dictdiffer/utils.py]: I would move this near the top (imports -> classes -> functions).

File: dictdiffer/utils.py [hunk 3]
```diff
@@ -11,10 +11,14 @@
 
 import sys
 
-from ._compat import izip_longest, num_types, string_types
+from ._compat import (MutableMapping, MutableSequence, izip_longest, num_types,
+                      string_types)
 
 EPSILON = sys.float_info.epsilon
 
+DICT_TYPES = (MutableMapping, )
```
Comment (iulianav) [inline dictdiffer/utils.py]: I feel like this is duplicated knowlege as it is present [here](https://github.com/inspirehep/dictdiffer/pull/1/files#diff-d8d1353909cc9f4e93181a1e680a2ae9R35) as well. Maybe we should move DICT_TYPES and LIST_TYPES in a centralised place and import from there?

File: dictdiffer/utils.py [hunk 4]
```diff
@@ -271,3 +275,24 @@ def are_different(first, second, tolerance):
         return abs(first-second) > tolerance * max(abs(first), abs(second))
     # we got different values
     return True
+
+
+def strip_removed_objects(obj):
+    """Strip all the values that are instance of ``RemovedObject``."""
```
Comment (iulianav) [inline dictdiffer/utils.py]: 'instances' ?

File: tests/test_utils.py [hunk 5]
```diff
@@ -144,3 +146,64 @@ def test_nested_hash(self):
         nested_hash((1, 2, 3))
         nested_hash(set([1, 2, 3]))
         nested_hash({'foo': 'bar'})
+
+
+    def test_strip_removed_objects(self):
```
Comment (iulianav) [inline tests/test_utils.py]: I would break this test in separate ones that have meaningful names.

File: dictdiffer/utils.py (hunk 1 above)
Comment (david-caro) [reply dictdiffer/utils.py]: That you don't need to `strip_removed_objects` of the `value` if it's a `RemovedObject`, so you can skip this line if `isinstance(value, RemovedObject)`, either moving it inside the following if, or inverting the if and continuing:
```python
for ...:
    if isinstance(value, RemovedObject):
        continue

    clean_value = ..
    new_dict[key] = clean_value
```

File: tests/test_merge.py [hunk 6]
```diff
@@ -64,6 +64,26 @@ def test_continue_run_multiple_conflicts_per_patch(self):
             self.assertEqual(patch(m.unified_patches, lca),
                              expected_value)
 
+    def test_continue_run_multiple_conflicts_per_patch_two(self):
```
Comment (iulianav) [inline tests/test_merge.py]: This name needs to be changed. The 'two' is funny.

**Comment by drjova** [comment]
@david-caro done

File: dictdiffer/utils.py [hunk 7]
```diff
@@ -271,3 +279,21 @@ def are_different(first, second, tolerance):
...
+    if isinstance(obj, LIST_TYPES):
+        return [
+            strip_removed_objects(value)
+            for value in obj if not isinstance(value, RemovedObject)
+        ]
+    if isinstance(obj, DICT_TYPES):
+        new_dict = {}
+        for key, value in obj.items():
+            if isinstance(value, RemovedObject):
```
Comment (iulianav) [inline dictdiffer/utils.py]: I also wanted to ask if you can do this! Great!

File: tests/test_merge.py (hunk 6 above)
Comment (drjova) [reply tests/test_merge.py]: I didn't even noticed 👍

File: tests/test_utils.py (hunk 5 above)
Comment (drjova) [reply tests/test_utils.py]: actually that's a valid point, in general we should have used ``parametrize```

Comment (iulianav) [reply tests/test_utils.py]: Yes, that sounds even better!

File: dictdiffer/utils.py (hunk 3 above)
Comment (drjova) [reply dictdiffer/utils.py]: good point but we cannot do it, ``__init__`` imports from ``utils``

**Pull request merged**